CIP-0021/README.md

```diff
@@ -94,34 +104,36 @@ Since multiassets (`policy_id` and `asset_name`) are represented as maps, both n
 
 #### Certificates
 
-Certificates of type `genesis_key_delegation` and `move_instantaneous_rewards_cert` are not supported and must not be included.
+Certificates of the following types are not supported and must not be included:
+* `genesis_key_delegation`
+* `move_instantaneous_rewards_cert`
+* `stake_vote_deleg_cert`
+* `stake_reg_deleg_cert`
+* `vote_reg_deleg_cert`
+* `stake_vote_reg_deleg_cert`
 
 If a transaction contains a pool registration certificate, then it must not contain:
```

Comment:
why do we not add any protections like this around constitutional committee keys or DRep keys

Reply:
The main reason for pool registration certificate restrictions is that pool owners are signing them with stake keys, which also sign other elements of the transaction (e.g. withdrawals and other certificates), and that's a security risk: the transactions are not created by them and so they might contain, from their viewpoint, third-party elements --- they can't easily distinguish whether a key hash used in a withdrawal belongs to them or not, and a HW wallet does not see the whole tx at once, so cannot warn them that the same stake key hash is used in several places. 

With DRep or constitutional committee keys, only votes are signed, nothing else AFAIK.

Reply:
> With DRep or constitutional committee keys, only votes are signed, nothing else AFAIK.

I have not tested this, but when you do a DRep deregistration, you could also sneak in a vote too. Not sure what the protocol is doing with that.

Reply:
DRep keys are used for votes, registration certs, update certs and resignation certs.

CC Hot keys are only used for voting.
CC Cold keys are used for authorising hot cert and resign cert.

Reply:
@Ryun1 do you think its worth to check if there is a vote together with a cert in the transaction?

Reply:
@gitmachtl 

to follow the precedent of how stake key signing is handled, yes it makes sense for the check.

But practically speaking, I don't think this is a significant issue, so IMO it is probably fine.
It is quite easy for a DRep to re-register.

Reply:
@Ryun1 i was more thinking in a way that if online services provide the feature to easy register as a DRep, if they could sneak in a vote for there proposal f.e. and the user would not see that directly.

Reply:
yes this would be easy to do over a dApp.
but of course on the hardware wallet device it still shows the vote to the user, at signing time.